src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -751,6 +751,13 @@ public function deprecateParameter(string $name, string $package, string $versio
      */
     public function compile(bool $resolveEnvPlaceholders = false)
     {
+        foreach ($this->extensions as $extension) {
+            // @deprecated remove if statement in Symfony 7
+            if (method_exists($extension, 'build') && \is_callable([$extension, 'build'])) {
+                $extension->build($this);
```

Comment:
Shouldn't this happen in Kernel::prepareContainer instead?
This would give the highest precedence to the App\Kernel, which looks desired to me.

Reply:
This feature is meant to be used when using the DI component standalone (like in `phpdocumentor/guides`), so that doesn't work.

If the precedence is important, we can maybe call this from `Container::registerExtension()`? In that case, the precedence would be Kernel, Bundle, Extension ?

Reply:
I think that'd be better yes.